src/snoowrap.js

```diff
@@ -639,6 +639,45 @@ const snoowrap = class snoowrap {
     return this._get({uri: 'api/saved_categories'}).get('categories');
   }
 
+  /**
+   * @param {string} query a query to search
+   * @param {Object} [config] config object
+   * @returns {Promise} A promise with array os subreddits starting with that query
+   * @example
+   * const config = {
+   *  exact:false,
+   *  include_over_18:false,
+   *  include_unadvertisable:false
+   * };
+   * const {subreddits} = await r.searchSubreddit("learn", config);
+   * => [
+   * {
+   *   active_user_count: 1705
+   *   allow_chat_post_creation: null
+   *   allow_images: false
+   *   icon_img: ""
+   *   key_color: ""
+   *   name: "learnprogramming"
+   *   subscriber_count: 1084731
+   * },
+   * {
+   *   active_user_count: 126
+   *   allow_chat_post_creation: null
+   *   allow_images: true
+   *   icon_img: ""
+   *   key_color: ""
+   *   name: "learnjavascript"
+   *   subscriber_count: 76606
+   * }
+   * ]
+   */
+  searchSubreddit (query, config = {exact: false, include_over_18: false, include_unadvertisable: false}) {
+    if (query.length > 50) {
```

Comment:
Does the endpoint fail on length of 50 or length of 49? Test indicates greater than or equal, if-statement indicates greater than

Reply:
hmmm the endpoint isn't actually fail. It simply returning an empty array which can also mean there is no results match `"x".repeat(51)`.

Reply:
Oh, if the endpoint doesn't return an error we shouldn't throw an error either, better to remove it